src/loader/endpoints/get_inloco_cities.py

```diff
@@ -1,51 +1,169 @@
+            state_cities_inloco = self.states[state_num_id]
+            self.correct_names(state_cities_inloco, state_cities_ref, state_num_id)
+        final_df = df.copy(deep=True)
+        final_df["city_name"], final_df["city_id"], final_df["state_num_id"] = zip(
+            *final_df.apply(self.finalize_df, axis=1)
+        )
+        self.final_df = final_df
+
+    def get_clean_df(self):
+        return self.final_df
+
+    def sort_df(self, state_num_id, city_name):
+        idx = self.states[state_num_id].searchsorted(city_name)
+        if (
+            idx >= len(self.states[state_num_id])
+            or self.states[state_num_id][idx] != city_name
+        ):
+            self.states[state_num_id] = np.concatenate(
+                (
+                    self.states[state_num_id][:idx],
+                    [city_name],
+                    self.states[state_num_id][idx:],
+                )
+            )
 
-# Adds the city_id column on inlocos data (must be after cleaning)
-def insert_city_id(in_df):
-    df = in_df
-    ids = []
-    for index, row in df.iterrows():
-        state_num_id = states_table.loc[
-            states_table["state_name"] == row["state_name"]
-        ]["state_num_id"].values[0]
-        try:
-            city_id = cities_table.loc[
-                (cities_table["state_num_id"] == state_num_id)
-                & (cities_table["city_name"] == row["city_name"])
-            ]["city_id"].values[0]
-        except:
-            print(
-                states_table.loc[states_table["state_name"] == row["state_name"]][
-                    "state_num_id"
-                ].values
+    def sort_ref(self, state_num_id, city_name, city_id):
+        idx = self.states_references[state_num_id][:, 0].searchsorted(
+            city_name
+        )  # uses only the first item
+        if (
+            idx >= len(self.states_references[state_num_id])
+            or self.states_references[state_num_id][idx][0] != city_name
+        ):
+            self.states_references[state_num_id] = np.concatenate(
+                (
+                    self.states_references[state_num_id][:idx],
+                    np.array([[city_name, str(city_id)]]),
+                    self.states_references[state_num_id][idx:],
+                )
             )
-            print(
-                cities_table.loc[cities_table["city_name"] == row["city_name"]],
-                row["city_name"],
-                row["state_name"],
+
+    def convert_state_name_to_num_id(self, state_name):
+        state_index = np.searchsorted(self.states_names, state_name)
+        return self.states_num_ids[state_index]
+
+    def solve_word(self, wrong_word, correct_word, lev_dis):
+        if lev_dis <= 2:
+            return correct_word
+        else:
+            try:
+                return self.corrections["inloco"][wrong_word]["correct_name"]
+            except:
+                print("Not found in corrections " + wrong_word)
```

Comment:
Should this raise an error? Use our logger module to assing a 'priority' to it.

Just, 

`from loader.logger import logger`

and then you can use the logger

`logger.debug`, `logger.info`, `logger.warning`, logger.error`, depending on how bad it is.

It seems like it is a `warning`

Reply:
In theory no because its a mistake we can afford to make, since its only for a few small cities, but I will add it to the logger.